### ISSUE TITLE
Unexpected UI when an active tool with UI is unloaded, runs out of charge, or when the player tries to sleep with the tool on

### ISSUE THREAD

**Issue by Qrox** [CleverRaven/Cataclysm-DDA#21093]
Related to #18033. There are these lines of code from `game::unload( item &it )`

```c++
    // Turn off any active tools
    if( target->is_tool() && target->active && target->ammo_remaining() == 0 ) {
        target->type->invoke( &u, target, u.pos() );
    }
```

The code tries to deactivate a tool by activating it. If the tool shows UI when activated, unexpected UI will show when the the tool is active, and is then unloaded, runs out of charge, or if the player tries to sleep with the tool on.

The last scenario is fixable by not showing the UI unless it's the player who is activating the item (see #21014). For the first two, that fix is not applicable. It can be worked around by not showing the UI unless the tool has remaining charge, but it will not work for items that have multiple functions both when charged and not charged (Is there any item of this kind?).

Affected tools:

- E-ink tablet PC
- Remote vehicle controller
- Multi cooker
- RC car
- RC control
- Torch (When running out of charge)
- Louisville Slaughterer (When running out of charge)

**Comment by Treah** [comment]
Is the E-ink Tablet the only device that has a UI when its activated that has the ability to stay active after you close the UI?

**Comment by Qrox** [comment]
A search for `uimenu` and `menu()` in iuse.cpp turned out these items: remote vehicle controller, multi cooker, RC car, and RC control. These items are also reproducing the bug in #18033.

The bug also happens when the charge runs out, which also happens to torch and Louisville Slaughterer.

**Comment by Night-Pryanik** [comment]
Fixed by #76657.
